core/src/state.rs

```diff
@@ -927,26 +893,25 @@ impl Drop for TransactionEventBuffer<'_> {
     }
 }
 
+impl StateBlock<'_> {
+    pub(crate) fn consensus_estimation(&self) -> Duration {
+        let Parameters {
+            block_time,
+            commit_time,
+            ..
+        } = self.world.parameters();
+
+        *block_time + (*commit_time / 2)
```

Comment:
That would also make sense to move to the `Parameters` struct itself:
```suggestion
        self.world.parameters().consensus_estimation()
```